docs/cache-server.md

```diff
@@ -0,0 +1,125 @@
+
+The second mode of operation is implemented as a separate informer within a shard. 
+Thanks to that a shard can implement a different resiliency strategy. 
+For example, it can tolerate the unavailability of the secondary during startup and become ready.
+
+### Running the server
+
+The cache server can be run as a standalone binary or as part of a kcp server.
+
+The standalone binary is in https://github.com/kcp-dev/kcp/tree/main/cmd/cache-server and can be run by issuing `go run ./cmd/cache-server/main.go` command.
+
+To run it as part of a kcp server, pass `--cache-url` flag to the kcp binary.
+
+### Client-side functionality
+
+In order to interact with the cache server from a shard, the https://github.com/kcp-dev/kcp/tree/main/pkg/cache/client 
+repository provides the following client-side functionality:
+
+[ShardRoundTripper](https://github.com/kcp-dev/kcp/blob/b739fa5b5c83fb2c43b631c6234264d5dd1fc6e4/pkg/cache/client/round_tripper.go#L54), 
+a shard aware wrapper around `http.RoundTripper`. It changes the URL path to target a shard from the context.
+
+[DefaultShardRoundTripper](https://github.com/kcp-dev/kcp/blob/b739fa5b5c83fb2c43b631c6234264d5dd1fc6e4/pkg/cache/client/round_tripper.go#L128) 
+is a `http.RoundTripper` that sets a default shard name if not specified in the context.
+
+For example, in order to make a client shard aware, inject the `http.RoundTrippers` to a `rest.Config`
+
+```
+import(
+cacheclient "github.com/kcp-dev/kcp/pkg/cache/client"
+"github.com/kcp-dev/kcp/pkg/cache/client/shard"
+)
+
+// adds shards awareness to a client with a default `shard.Wildcard` name.
+NewClusterForConfig(cacheclient.WithShardRoundTripper(cacheclient.WithDefaultShardRoundTripper(serverConfig.LoopbackClientConfig, shard.Wildcard)))
+
+// and then change the context when you need to access a specific shard and pass is when making a HTTP request
+ctx = cacheclient.WithShardInContext(ctx, shard.New("cache"))
+```
+
+### Authorization/Authentication
+
+Not implemented at the moment
+
+### Build-in resources
+
+Out of the box, the server supports the following resources:
+- `apiresourceschemas`
+- `apiexports`
+
+All those resources are stored in `system:cache:server` shard under `system:system-crds` cluster.
+
+### Adding new resources
+
+Not implemented at the moment.
+Only posting data explicitly is possible (e.g. POST against the extension endpoint).
+In the future, some form of automatic discovery of resources to install will be implemented.
+
+### Deletion of data
+
+Not implemented at the moment.
+Only deleting resources explicitly is possible.
+In the future, some form of automatic removal will be implemented.
+
+### Design details
+
+The cache server is implemented as the `apiextensions-apiserver`. 
+It is based on the same fork used by the kcp server, extended with shard support.
+
+Since the server serves as a secondary replica it doesn't support versioning, validation, pruning, or admission. 
```

Comment:
Do we expect this to be the case forever? How do we handle normal client-go SDK clients asking for specific versions and getting something else back from the cache server?

Reply:
Do we even support discovery on cache server? Will we?

Reply:
> Do we expect this to be the case forever? How do we handle normal client-go SDK clients asking for specific versions and getting something else back from the cache server?

versioning is implicit, imposed when shards add data to the cache.

Reply:
Are we certain about this?

Reply:
maybe @sttts could confirm that

Reply:
yes, there is expected to be a consensus (non-technically, but socially during development) what do store in the case and in which version (only one). Eventually, we have to think more about how to transition to a new version. For now the data there is intentionally basically untyped (= schema), i.e. no pruning, no validation, ... If we could, I would even like to avoid any CRD to be created before we can serve something, i.e. much more like KV stores work where you can write what you want and the clients have to agree on keys and values.